### PR TITLE
Update spiral_print.py

### PR DESCRIPTION
**Pull request by kondekarshubham123**
### Describe your change:

Added another neat approach for same question with explaination 

* [X] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [X] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: matrix/spiral_print.py [hunk 1]
```diff
@@ -7,6 +7,9 @@
 """
 
 
+from typing import List
```
Comment (CaedenPH) [inline matrix/spiral_print.py:10]:
```suggestion
```
You can just use `list` to keep consistency with the file
This repository also targets at python 3.10+

File: matrix/spiral_print.py [hunk 2]
```diff
@@ -75,8 +78,67 @@ def spiral_print_clockwise(a: list[list[int]]) -> None:
         print("Not a valid matrix")
         return
 
+# Other Easy to undersatnd Approach
+
+
+def spiral_traversal(matrix: List[List[int]]) -> List[int]:
```
Comment (CaedenPH) [inline matrix/spiral_print.py]:
```suggestion
def spiral_traversal(matrix: list[list[int]]) -> list[int]:
```

File: matrix/spiral_print.py [hunk 3]
```diff
@@ -75,8 +78,67 @@ def spiral_print_clockwise(a: list[list[int]]) -> None:
...
+                                                        [6],
+                                                        [7]
+                                    ])
+
+    Stage 5.
+    [1, 2, 3, 4, 8, 12, 11, 10, 9, 5] + spiral_traversal(  [[6, 7]
+                                    ])
+
+    Stage 6.
+    [1, 2, 3, 4, 8, 12, 11, 10, 9, 5, 6, 7] + spiral_traversal([])
```
Comment (CaedenPH) [inline matrix/spiral_print.py:132]:
```suggestion
    Stage 1.
    [1, 2, 3, 4] +   spiral_traversal([
        [8, 12], [7, 11], [6, 10], [5, 9]]
     ])

    Stage 2.
    [1, 2, 3, 4, 8, 12] + spiral_traversal([
        [11, 10, 9], [7, 6, 5]
    ])

    Stage 3.
    [1, 2, 3, 4, 8, 12, 11, 10, 9] + spiral_traversal([
        [5], [6], [7]
    ])

    Stage 4.
    [1, 2, 3, 4, 8, 12, 11, 10, 9, 5] + spiral_traversal([
        [5], [6], [7]
    ])

    Stage 5.
    [1, 2, 3, 4, 8, 12, 11, 10, 9, 5] + spiral_traversal([[6, 7]])

    Stage 6.
    [1, 2, 3, 4, 8, 12, 11, 10, 9, 5, 6, 7] + spiral_traversal([])
```

**Review by CaedenPH — COMMENTED** [review]
It seems `Stage 3.` and `Stage 4.` are the same. Is this intentional?

**Comment by kondekarshubham123** [comment]
Changes did as per mentioned, please review again @cclauss , @CaedenPH 

Regards, 
Shubham

**Comment by cclauss** [comment]
Please do not ask for reviews when tests are failing (❌ ) because we have ~200 pull requests and this is a volunteer activity.

**Comment by kondekarshubham123** [comment]
Okay I do understand.

File: matrix/spiral_print.py [hunk 4]
```diff
@@ -123,7 +123,9 @@ def spiral_traversal(matrix: list[list[int]]) -> list[int]:
     Stage 6.
     [1, 2, 3, 4, 8, 12, 11, 10, 9, 5, 6, 7] + spiral_traversal([])
     """
-    return matrix and list(matrix.pop(0)) + spiral_traversal(list(zip(*matrix))[::-1])
+    if matrix:
+        spiral_traverse_mat = list(matrix.pop(0)) + spiral_traversal(list(zip(*matrix))[::-1])
+        return spiral_traverse_mat
```
Comment (cclauss) [inline matrix/spiral_print.py]:
```suggestion
    matrix = [[]]
    return list(matrix.pop(0)) + spiral_traversal(list(zip(*matrix))[::-1])
```

Comment (kondekarshubham123) [reply matrix/spiral_print.py]: Ohh, from long time I was trying to resolve this only,

Thanks (^_^);

**Comment by cclauss** [comment]
=================================== FAILURES ===================================
________________ [doctest] matrix.spiral_print.spiral_traversal ________________
083 
084     >>> spiral_traversal([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
UNEXPECTED EXCEPTION: RecursionError('maximum recursion depth exceeded while calling a Python object')
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.8/x64/lib/python3.10/doctest.py", line 1350, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest matrix.spiral_print.spiral_traversal[0]>", line 1, in <module>
  File "/home/runner/work/Python/Python/matrix/spiral_print.py", line 127, in spiral_traversal
    return list(matrix.pop(0)) + spiral_traversal(list(zip(*matrix))[::-1])
  File "/home/runner/work/Python/Python/matrix/spiral_print.py", line 127, in spiral_traversal
    return list(matrix.pop(0)) + spiral_traversal(list(zip(*matrix))[::-1])
  File "/home/runner/work/Python/Python/matrix/spiral_print.py", line 127, in spiral_traversal
    return list(matrix.pop(0)) + spiral_traversal(list(zip(*matrix))[::-1])
  [Previous line repeated 961 more times]
RecursionError: maximum recursion depth exceeded while calling a Python object
/home/runner/work/Python/Python/matrix/spiral_print.py:84: UnexpectedException

**Comment by kondekarshubham123** [comment]
After running 
`python3 -m doctest -v matrix/spiral_print.py`
I'm getting this as op
```
2 items had no tests:
    spiral_print
    spiral_print.check_matrix
2 items passed all tests:
   1 tests in spiral_print.spiral_print_clockwise
   1 tests in spiral_print.spiral_traversal
2 tests in 4 items.
2 passed and 0 failed.
Test passed.
```

**Review by cclauss — APPROVED** [review]

**Pull request merged**